pkg/segment/structs/evaluationstructs.go

```diff
@@ -1533,6 +1533,52 @@ func (self *ValueExpr) EvaluateValueExprAsString(fieldToValue map[string]utils.C
 	return str, nil
 }
 
+func handleCaseFunction(self *ConditionExpr, fieldToValue map[string]utils.CValueEnclosure) (string, error) {
+
+	for _, cvPair := range self.ConditionValuePairs {
+		res, err := cvPair.Condition.Evaluate(fieldToValue)
+		if err != nil {
+			continue
+		}
+		if res {
+			val, err := cvPair.Value.EvaluateValueExprAsString(fieldToValue)
+			if err != nil {
+				return "", fmt.Errorf("handleCaseFunction: Error while evaluating value, err: %v", err)
+			}
+			return val, nil
+		}
+	}
+
+	return "", nil
+}
+
+func handleCoalesceFunction(self *ConditionExpr, fieldToValue map[string]utils.CValueEnclosure) (string, error) {
+	for _, valueExpr := range self.ValueList {
+		fields := valueExpr.GetFields()
+
+		skip := false
+		for _, field := range fields {
+			val, ok := fieldToValue[field]
+			// if a field has a NIL value in the expression we do not want that expression to be considered
+			if !ok || val.Dtype == utils.SS_DT_BACKFILL {
+				skip = true
+				break
+			}
+		}
+		if skip {
+			continue
+		}
```

Comment:
FYI: in Go you can label loops and then `continue` or `break` those loops even while you're inside an inner loop. See https://go.dev/play/p/HkQCdidkgcw for an example

But what you have with the `skip` variable is also fine